### DISCUSSION TITLE
Entrada no mercado de trabalho "avançado"

### DISCUSSION THREAD

**Discussion by dsantoro** [frontendbr/forum#625] [Discussão]
Bom dia,
Esse questionamento surgiu no twitter e o Matheus Lima sugeriu que fosse postado aqui.

Trabalho em uma agência a quase 5 anos e quem já trabalhou em agência sabe a correria que é, mas ao meu ver isso é uma coisa boa pois o aprendizado aqui é constante e rápido, porém não podemos nos dar ao luxo de usar coisas mais legais como React, ou alguma coisa em Node, etc.

Estou tentando entrar nesse mercado a algum tempo mas notei que ele não é muito simpático com os novos. Sempre pedem gente com uma experiência pelo menos.

Entrei no mercado de frontend através de um estágio quando cursava a faculdade mas esse artifício não vai funcionar mais para mim rs.

Então, alguma dica para mim e para outros com o mesmo problema que eu? 

Abraços!

**Comment by yanmagale** [comment]
@dsantoro então cara, vivi uma situação muito parecida com a sua ... 

# Sobre o ecossistema em si
O que acontece é que, pelo próprio modelo de negócio das agências em geral, as coisas são muito corridas, são sempre para antes de antes de ontem, então realmente não sobra tempo para se trabalhar com estas novas tecnologias(embora eu acredito que uma boa conversa com o seu time e com seus líderes, mostrando sobre os benefícios daquela tecnologia que você quer utilizar, possa ser bem efetivo :smile: ).

Então nesse cenário realmente é um pouco complicado de você conseguir inovar(claro que não é em todas as agências que isso acontece)

# Sobre as mudanças

O que eu fiz foi "alinhar" comigo mesmo as minhas expectativas sobre o que eu queria para mim: poxa, eu quero ficar sempre fazendo as mesmas coisas que eu faço aqui na agência, ou eu quero aprender mais, ir para um próximo nível e participar de projetos desafiadores, onde eu tenha a possibilidade de participar e decidir juntamente com o time sobre a melhor stack para aquele cenário de projeto?

Comecei a participar das comunidades e eventos sobre as tecnologias que eu queria me aprofundar(no caso que vc citou, seria sobre React e Node por exemplo), participar de slack das comunidades, ver posts no medium e claro, comecei a utilizar estas tecnologias, em projetos pessoais. Quanto mais você a utilizar, mais vai evoluir. E como consequência, você acaba tendo acesso as vagas nas empresas que utilizam aquelas tecnologias.

Meu conselho seria você alinhar esta expectativa, participar do ecossistema daquela linguagem/tecnologia que você queira utilizar e dar tempo para que as oportunidades possam surgir.

Espero ter ajudado :smile:

**Comment by matheusml** [comment]
Oi @dsantoro 

Então, vou tentar resumir de uma forma mais objetiva o que eu faria se estivesse no seu lugar.

**1. Network**

Esse é o momento ideal para procurar na sua lista de conhecidos quem está trabalhando com React, ou quem conhece alguém que trabalhe.
Por isso é tão importante fazer _network_. Nesses momentos de mais desespero, ele pode trazer uma certa tranquilidade.

Das 5 empresas que eu trabalhei, 4 foram por indicação, só pra dar uma ideia da importância do _network_. 

Lembro quando Angular.js estava começando no Brasil e eu apostava bastante nele. Fui atrás de alguns colegas que estavam trabalhando em empresas com a tecnologia, pedi pra ser indicado, 2 semanas depois já estava trabalhando com Angular.

**2. Sites com vagas**

Além disso, você pode procurar por alguns sites de vagas. São diversas opções para filtrar por cidade + tecnologia.
Por exemplo: se você for de São Paulo, pode fazer um filtro com São Paulo + React. Tenho certeza que diversas empresas vão aparecer no filtro.

**3. Estude**

Em paralelo, enquanto você procura as vagas, vá estudando o React. Isso vai ser bom por dois motivos. Primeiro porque vai te ajudar a ir melhor nas entrevistas que você está procurando, e segundo porque depois que você passar na entrevista, o ideal é já chegar na empresa pelo menos sabendo o básico do React.

Espero ter ajudado,
Abraços!

**Comment by felipefialho** [comment]
Olá @dsantoro, beleza?

Tenho esse texto de 2015, na qual dou algumas dicas para as pessoas estudarem e colocarem na prática seus estudos para se aproximar do mercado. Da uma olhada.

[Aproximando seus estudos de cenários reais](https://www.felipefialho.com/blog/2015/aproximando-seus-estudos-de-cenarios-reais)

Paralelo a isso, pode tentar fazer [testes open source de algumas empresas](https://github.com/LFeh/frontend-challenges), além de estudar, vai que faz um teste muito foda e ganha uma chance?

**Comment by dsantoro** [comment]
BASTANTE legal esse repositório de testes de grandes empresas @LFeh vou certamente dar uma olhada nesses testes!

**Comment by rubenmarcus** [comment]
Eu to nessa também @dsantoro. Mas empresa grande tambem exige bastante. A diferença que geralmente a estrutura é melhor.

Acho que a idéia central é você sempre estudar e aplicar os conhecimentos em projetos pessoais,open source ou algum projeto seu como traduzir conteúdo , colaborar em eventos não sei.

Eu tenho quase 6 anos de front-end, e perdi muito tempo sendo o cara que entrega projetos, comi bola. Eu separei esses 6 meses até Janeiro/2018 pra atualizar meus skills técnicos, eu to trampando atualmente como freelancer em projetos de implementação de UI para ter uma renda, mas pretendo ano que vem estar estruturado pra entrar em vagas bacanas.




Não sei se é um feedback bom, Mas muita empresa difere o discurso na hora da contratação e na hora do trabalho em si.Por isso não espere nada. Eu sei que as vezes é foda, tipo você trabalhar de 8- 10 horas por dia no minimo, ai depois ter que ir pra casa estudar, isso se não tiver mulher, filhos e outras obrigações, mas não tem como é um investimento tem que ser feito, o lance das empresas é que tem empresa que diz que vai valorizar o profissional, fazer treinamentos, pagar eventos, etc etc, e eu me vejo muito mais como um profissional que investe nos projetos pessoais e no open source, do que acreditar em "empresas", sei que existe MUITAS empresas boas, mas também tem ruins..

Não é só também skills tecnicos mas cultura, lifestyle , tipo lance de horarios, beneficios é foda,

eu tb tentei essa abordagem em diversas empresas, muitas vezes era cobrado por tempo parado que ficava, quando propunha adotar algo ou fazer outra coisa, a empresa dizia que não iria usar certa tecnologia por não ter dominio sobre ela ou etc..

**Comment by rubenmarcus** [comment]
@dsantoro Cara, eu to indo pra Índia em Outubro e pretendo ver a cena de tecnologia lá, estou indo pra fotografar, mas vou também tentar programar todos os dias, se você quiser fazer algum projeto juntos, em react, react-native, estamos aí. 

no [Upwork](https://www.upwork.com) Tem uns testes de conhecimento front-end. Eu não sei muitos sites mais que tenham, mas vale a pena procurar

**Comment by lucianobarauna** [comment]
@dsantoro 
Estou passando por uma situação parecida. Na agência onde trabalhei durante 6 anos o front-end quase não pegava no javascript e quem tinha a responsabilidade era o back-end. 

A minha estratégia foi começar a estudar igual um louco JS para melhorar e quando chega-se a minha hora eu arriscaria e trocaria de emprego. Tem 4 meses que mudei de emprego, consegui ele por networking e estou trabalhando com React. 

Hoje gerencio o design e faço o front-end de alguns projetos mais estou metendo muito mais a mãe em javascript e continuo estudando ;-)

Cara o que vc está fazendo é o certo saia da zona de conforto e queira algo melhor ;-)

**Comment by adeonir** [comment]
@lucianobarauna 
Estou no mesmo caminho. Sempre trabalhei com design + HTML/CSS. Esse ano estou focado em aprender JS para tentar um novo emprego.

**Comment by lucianobarauna** [comment]
@adeonir 
Cara fico muito feliz quando vejo que doidos iguais a mim ahuiehaiuehae

**Comment by didiraja** [comment]
@dsantoro nossa situação é tão parecida que só faltou você ser do RJ hahah

Fui querer entrar no meio de front-end propriamente dito e como você, percebi que as vagas são apenas de pleno pra cima. Até quando falam Júnior, é um junior com quase o mesmo conhecimento de um Pleno.

O que me(nos) resta então é estudar. Estudar pra aprender tudo que pedem e então por participar de verdade dos processos seletivos.

Eu mesmo, independente desse mercado, estava um pouco defasado e to correndo pra tirar esse atraso, pra então ficar equiparado ao mercado. Porém, admito que tá sendo bem prazeroso aprender coisas novas.

No mais, boa sorte pra nós. :)
